src/rmutil/periodic.c

```diff
@@ -40,7 +40,9 @@ static void *rmutilTimer_Loop(void *ctx) {
       if (RedisModule_GetThreadSafeContext) rctx = RedisModule_GetThreadSafeContext(NULL);
 
       // call our callback...
-      tm->cb(rctx, tm->privdata);
+      if (!tm->cb(rctx, tm->privdata)) {
+        break;
+      }
```

Comment:
Needs to be merged upstream (https://github.com/RedisLabs/RedisModulesSDK/tree/master/rmutil)

Reply:
was not aware of that  ...